pkg/messaging/message.go

```diff
@@ -11,3 +21,44 @@ type Message struct {
 func (m *Message) IsBroadcast() bool {
 	return m.To == ""
 }
+
+type controlPayload struct {
+	Type string `json:"type"`
+	Peer string `json:"peer"`
+}
+
+func NewPeerDisconnected(logger logging.Logger, peerUID string) Message {
+	payload := controlPayload{
+		Type: ctrlDisconnected,
+		Peer: peerUID,
+	}
+
+	jsonPayload, err := json.Marshal(payload)
+	if err != nil {
+		logger.Error("failed to marshal control payload", logging.Fields{"type": "peer_disconnected", "peer": peerUID})
```

Comment:
This error handling is not good. If there's an error, this method shouldn't return a Message. It now always returns a message, but what will be the payload if it failed to marshal? I thin it needs to return Message or error and do no logging (this will resolve the issue with having to pass logger here). Callers will then need to handle the error and log if necessary.

Reply:
Callers will also be able to provide more context to the log, like room id.